CMakeLists.txt

```
@@ -306,6 +306,16 @@ set(BOUT_SOURCES
   ${CMAKE_CURRENT_BINARY_DIR}/include/bout/version.hxx
   )
 
+
+ADD_CUSTOM_COMMAND( OUTPUT ${CMAKE_CURRENT_SOURCE_DIR}/src/field/generated_fieldops.cxx
+  COMMAND python gen_fieldops.py --filename generated_fieldops.cxx.tmp
+  COMMAND clang-format generated_fieldops.cxx.tmp -i
+  COMMAND mv generated_fieldops.cxx.tmp generated_fieldops.cxx
```

We can also make this lovely and portable too:

```suggestion
  COMMAND ${CMAKE_COMMAND} -E rename generated_fieldops.cxx.tmp generated_fieldops.cxx
```

https://cmake.org/cmake/help/latest/manual/cmake.1.html#run-a-command-line-tool

---

Sorry for not replying earlier.

`mv` is defined by POSIX, thus only non-POSIX complient OS would be affected by that.
I think `mv` is lovely :smiley: - calling cmake to move a file however feels like overkill.
As this would only cause issues on an OS that is intentionally non-POSIX compliant thus I disagree with aiming for portability to such a degree. I think such an OS shouldn't be catered for, and I would prefer we wouldn't support that. But the LGPL gives you every right to ignore my preference :+1: